### PR TITLE
Server ESM port: Basic imports 

### PR DESCRIPTION
**Pull request by eliandoran**
See https://github.com/TriliumNext/Notes/pull/241

### CODE REVIEW

File: spec/search/search.spec.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-import becca = require('../../src/becca/becca');
-// const { NoteBuilder, findNoteByTitle, note } = require("./becca_mocking");
-import becca_mocking = require('./becca_mocking');
+import searchService from "../../src/services/search/services/search.js";
+import BNote from "../../src/becca/entities/bnote.js";
+import BBranch from "../../src/becca/entities/bbranch.js";
+import SearchContext from "../../src/services/search/search_context.js";
+import dateUtils from "../../src/services/date_utils.js";
+import becca from "../../src/becca/becca.js";
+// const { NoteBuilder, findNoteByTitle, note } = require('./becca_mocking');
```
Comment (JYC333) [inline spec/search/search.spec.ts:7]: Can be removed?

**Review by JYC333 — CHANGES_REQUESTED** [review]
I didn't go through all the files, but I found most of import changes adding `js` at the end. But you already rename the file to `ts`. Like `import etapi from "../support/etapi.js";`, but you have file `../support/etapi.ts`. Could you fix this first?

**Comment by eliandoran** [comment]
> I didn't go through all the files, but I found most of import changes adding `js` at the end. But you already rename the file to `ts`. Like `import etapi from "../support/etapi.js";`, but you have file `../support/etapi.ts`. Could you fix this first?

This is actually a standard practice in ES modules. An extension has to be added in order to be compliant with import.

If we use .ts instead of .js it fails with:

> An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.ts(5097)

From what I see in https://www.typescriptlang.org/tsconfig/#allowImportingTsExtensions, this would only work in case we are using a bundler, however this is not the case for the server.

File: spec/search/parens.spec.ts [hunk 2]
```diff
@@ -1,24 +1,26 @@
-const handleParens = require('../../src/services/search/services/handle_parens');
+import handleParens from "../../src/services/search/services/handle_parens";
```
Comment (JYC333) [inline spec/search/parens.spec.ts:1]: Then should all file imports have the extension?

Comment (eliandoran) [reply spec/search/parens.spec.ts:1]: @JYC333 , that would be correct, yes. Initially the `tsconfig` was improperly configured, causing these issues to not show up as errors.

They are handled in: https://github.com/TriliumNext/Notes/pull/268

**Pull request merged**